File: angular/src/app/shared/helpers/date.helper.dart
class DateHelper {
  static List<String> monthsNames = [
    "",
    "JAN",
    "FEV",
    "MAR",
    "ABR",
    "MAI",
    "JUN",
    "JUL",
    "AGO",
    "SET",
    "OUT",
    "NOV",
    "DEZ"
  ];

  static String formatDateForAPI(String date) {
    List<String> dateParts = date.split("/");

    return "${dateParts[2]}-${dateParts[1]}-${dateParts[0]}";
  }

  static String formatHour(int seconds) {
    Duration duration = Duration(seconds: seconds);
    return "${duration.inHours.toString().padLeft(2, '0')}:${duration.inMinutes.remainder(60).toString().padLeft(2, '0')}:${(duration.inSeconds.remainder(60).toString().padLeft(2, '0'))}";
  }
}

File: react/public/helpers/date.helper.ts
export class DateHelper {
  static monthsNames: string[] = [
    "JAN",
    "FEV",
    "MAR",
    "ABR",
    "MAI",
    "JUN",
    "JUL",
    "AGO",
    "SET",
    "OUT",
    "NOV",
    "DEZ",
  ];

  static formatDateToDatabase = (date: string): string => {
    if (date.length < 10) {
      return "";
    }

    return `${date.slice(6, 10)}-${date.slice(3, 5)}-${date.slice(0, 2)}`;
  };
}
